apps/frontend/src/components/Inbox/InboxAddItem.tsx

```diff
@@ -0,0 +1,126 @@
+"use client"
+
+import React, { useState, useEffect, useRef } from "react"
+
+import { PlusIcon } from "@radix-ui/react-icons"
+
+import { useAuth } from "@/src/contexts/AuthContext"
+import useInboxStore from "@/src/lib/store/inbox.store"
+
+export const InboxAddItem: React.FC = () => {
+  const { session } = useAuth()
+
+  const [addingItem, setAddingItem] = useState(false)
+  const textareaRefTitle = useRef<HTMLTextAreaElement>(null)
+  const textareaRefDescription = useRef<HTMLTextAreaElement>(null)
+  const [title, setTitle] = useState("")
+  const [description, setDescription] = useState("")
+  const [date, setDate] = React.useState<Date | undefined>(new Date())
+  const [selectedPages, setSelectedPages] = React.useState<string[]>([])
+
+  const { addItem } = useInboxStore()
+
+  useEffect(() => {
+    const textarea = textareaRefTitle.current
+    if (textarea) {
+      textarea.style.height = "auto"
+      textarea.style.height = `${textarea.scrollHeight}px`
+    }
+  }, [title])
+
+  useEffect(() => {
+    const textarea = textareaRefDescription.current
+    if (textarea) {
+      textarea.style.height = "auto"
+      textarea.style.height = `${textarea.scrollHeight}px`
+    }
+  }, [description])
+
```

Comment:
This is not the best way I guess, These useEffects will run everytime a user presses a key, which will cause re-renders. You can use shadcn for this

```typescript
export function TextareaDemo() {
  const textareaRef = useRef<HTMLTextAreaElement>(null)

  useEffect(() => {
    if (textareaRef.current) {
      autosize(textareaRef.current)
    }

    return () => {
      if (textareaRef.current) {
        autosize.destroy(textareaRef.current)
      }
    }
  }, [])

  return (
    <Textarea
      ref={textareaRef}
      placeholder="Type your message here."
      className="resize-none overflow-hidden bg-transparent focus:outline-none focus:ring-0 focus:border-transparent border-none shadow-none"
    />
  )
}
``` 


Refer to this example -> [Ref](https://v0.dev/chat/jkLlmP5tKZL)
https://ui.shadcn.com/docs/components/textarea

Reply:
where is this autosize coming from?

Reply:
[Try This](https://v0.dev/chat/jkLlmP5tKZL?b=b_AhgrosqLMt6)

Reply:
i tried to do this way but the behavior is very different, look:

https://github.com/user-attachments/assets/977dd77a-529f-458f-9a3c-7f29c29c4bd3

Reply:
In that case just add a comment something like **TODO: Fix the textarea** 
I'll pick it up later

Reply:
ok